micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -54,15 +54,20 @@ impl NVSStorage {
         let mut nvs = self.nvs.borrow_mut();
         Ok(nvs.set_str(key, string)?)
     }
+    fn has_str(&self, key: &str) -> Result<bool, NVSStorageError> {
+        let nvs = self.nvs.borrow();
+        Ok(nvs.str_len(key)?.is_some())
+    }
```

Comment:
Does this make `has_key` obsolete? I don't see it used anywhere else and the errors from `has_str` would be a superset of errors from `has_key`